packages/financial-templates-lib/src/price-feed/InsuredBridgePriceFeed.ts

```diff
@@ -0,0 +1,172 @@
+    const parsedAncillaryData: any = parseAncillaryData(ancillaryData);
+    const matchedRelay = matchedRelays.filter(
+      (relay: Relay) =>
+        relay.depositId === parsedAncillaryData.depositId &&
+        relay.depositTimestamp === parsedAncillaryData.depositTimestamp &&
+        relay.recipient.substr(2).toLowerCase() === parsedAncillaryData.recipient &&
+        relay.sender.substr(2).toLowerCase() === parsedAncillaryData.l2Sender &&
+        relay.l1Token.substr(2).toLowerCase() === parsedAncillaryData.l1Token &&
+        relay.amount === parsedAncillaryData.amount.toString() &&
+        relay.slowRelayFeePct === parsedAncillaryData.slowRelayFeePct.toString() &&
+        relay.instantRelayFeePct === parsedAncillaryData.instantRelayFeePct.toString() &&
+        relay.quoteTimestamp === parsedAncillaryData.quoteTimestamp &&
+        relay.realizedLpFeePct === parsedAncillaryData.realizedLpFeePct.toString() &&
+        relay.slowRelayer.substr(2).toLowerCase() === parsedAncillaryData.slowRelayer &&
+        relay.depositContract.substr(2).toLowerCase() === parsedAncillaryData.depositContract
+    );
+
+    // TODO: Do we need to handle the case where all of these params are matched and matchedRelay.length > 1?
+    if (matchedRelay.length === 0) {
+      this.logger.debug({
+        at: "InsuredBridgePriceFeed",
+        message: "No relay event found for price request time",
+        priceRequestTime: time,
+      });
+      return this.toBNWei(isRelayValid.No);
+    }
+
+    const relay = matchedRelay[0];
+
+    // Validate that relay matches a deposit struct exactly.
+    const matchedDeposits = this.deposits.filter(
+      (deposit: Deposit) =>
+        deposit.depositId === relay.depositId &&
+        deposit.depositHash === relay.depositHash &&
+        deposit.timestamp === relay.depositTimestamp &&
+        deposit.sender === relay.sender &&
+        deposit.recipient === relay.recipient &&
+        deposit.l1Token === relay.l1Token &&
+        deposit.amount === relay.amount &&
+        deposit.slowRelayFeePct === relay.slowRelayFeePct &&
+        deposit.instantRelayFeePct === relay.instantRelayFeePct &&
+        deposit.quoteTimestamp === relay.quoteTimestamp
+    );
+    if (matchedDeposits.length > 1) {
+      this.logger.error({
+        at: "InsuredBridgePriceFeed",
+        message: "TODO: Handle multiple deposits associated with same relay attempt",
+      });
+      throw new Error("TODO: Handle multiple relays for same price request timestamp");
+    } else if (matchedDeposits.length === 0) {
+      this.logger.debug({
+        at: "InsuredBridgePriceFeed",
+        message: "No deposit event matching relay attempt",
+      });
+      return this.toBNWei(isRelayValid.No);
+    }
+
+    // Validate relays proposed realized fee percentage.
+    const expectedRealizedFeePct = this.l1Client.calculateRealizedLpFeesPctForDeposit(/* matchedDeposits[0] */);
```

Comment:
Why is this commented out? Because this function doesn't take a parameter yet?

Reply:
Yep exactly! But this is the probably interface